internal/dag/builder.go

```diff
@@ -518,6 +521,21 @@ func (b *Builder) computeHTTPProxy(proxy *projcontour.HTTPProxy) {
 			svhost.Secret = sec
 			svhost.MinProtoVersion = annotation.MinProtoVersion(proxy.Spec.VirtualHost.TLS.MinimumProtocolVersion)
 
+			// If FallbackCertificate is enabled, but no cert passed, set error
+			if proxy.Spec.VirtualHost.TLS.EnableFallbackCertificate {
```

Comment:
Is it the expectation that virtual host which opted for fallback certificate in HttpProxy has to set proxy.Spec.VirtualHost.TLS.secretName also ?

Reply:
Currently, the design is you still need to specify the secretname for each vhost to enable the fallbackCertificate. With `TLSCertificateDelegation`, you're able to make a "default" certificate available to specific, or all namespaces, so that functionality which exists in other controllers can still be utilized here with Contour. 

The main goal of the fallback certificate is to handle the use-case where requests are received that do not have SNI on them.